src/wallet/wallet.cpp

```diff
@@ -251,11 +259,18 @@ bool CWallet::AddWatchOnly(const CScript &dest)
 {
```

Comment:
I'd significantly prefer to not expose this version, making it explicit to callers that the key's nCreateTime is being set to 0 (which, after this PR, implies the wallet's nTimeFirstKey is set to 1).